constraint/pkg/apis/templates/v1/constrainttemplate_types_test.go

```diff
@@ -0,0 +1,256 @@
+// that we require for v1 CRD support
+func TestValidationVersionConversionAndTransformation(t *testing.T) {
+	trueBool := true
+	testCases := []struct {
+		name  string
+		v     *Validation
+		exp   *templates.Validation
+		error bool
+	}{
+		{
+			name: "Two deep properties, LegacySchema=true",
+			v: &Validation{
+				LegacySchema: true,
+				OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+					Properties: map[string]apiextensionsv1.JSONSchemaProps{
+						"message": {
+							Type: "string",
+						},
+						"labels": {
+							Type: "array",
+							Items: &apiextensionsv1.JSONSchemaPropsOrArray{
+								Schema: &apiextensionsv1.JSONSchemaProps{
+									Type: "object",
+									Properties: map[string]apiextensionsv1.JSONSchemaProps{
+										"key":          {Type: "string"},
+										"allowedRegex": {Type: "string"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			exp: &templates.Validation{
+				OpenAPIV3Schema: &apiextensions.JSONSchemaProps{
+					XPreserveUnknownFields: &trueBool,
+					Properties: map[string]apiextensions.JSONSchemaProps{
+						"message": {
+							Type: "string",
+						},
+						"labels": {
+							Type: "array",
+							Items: &apiextensions.JSONSchemaPropsOrArray{
+								Schema: &apiextensions.JSONSchemaProps{
+									Type:                   "object",
+									XPreserveUnknownFields: &trueBool,
+									Properties: map[string]apiextensions.JSONSchemaProps{
+										"key":          {Type: "string"},
+										"allowedRegex": {Type: "string"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			error: false,
+		},
+		{
+			name: "Two deep properties, LegacySchema=false",
+			v: &Validation{
+				LegacySchema: false,
+				OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+					Properties: map[string]apiextensionsv1.JSONSchemaProps{
+						"message": {
+							Type: "string",
+						},
+						"labels": {
+							Type: "array",
+							Items: &apiextensionsv1.JSONSchemaPropsOrArray{
+								Schema: &apiextensionsv1.JSONSchemaProps{
+									Type: "object",
+									Properties: map[string]apiextensionsv1.JSONSchemaProps{
+										"key":          {Type: "string"},
+										"allowedRegex": {Type: "string"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			exp: &templates.Validation{
+				OpenAPIV3Schema: &apiextensions.JSONSchemaProps{
+					Properties: map[string]apiextensions.JSONSchemaProps{
+						"message": {
+							Type: "string",
+						},
+						"labels": {
+							Type: "array",
+							Items: &apiextensions.JSONSchemaPropsOrArray{
+								Schema: &apiextensions.JSONSchemaProps{
+									Type: "object",
+									Properties: map[string]apiextensions.JSONSchemaProps{
+										"key":          {Type: "string"},
+										"allowedRegex": {Type: "string"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			error: false,
```

Comment:
We should remove the "error" config if we're not going to use it.

Reply:
I've removed this one and its equivalents from the transformation tests across the three CT version packages